pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
@bgrant0607 I see your point. The issue is that additions made by the user should take precedence over additions made by automated processes. Without special case logic, however, which probably doesn't belong here, we need to make it a blanket override, not something that applies only to certain properties, such as, say, default values.

However, it sounds like we want changes and deletions made to user supplied configuration by automated processes to cause update errors if conflicting additions, changes or deletions are made by the user, and we want to let the user force the api server to accept a conflicting update using a parameter, expressed for example as a --force flag on kubectl apply. Is this correct? 

Do we have good examples of changes and deletions made by automated processes that might conflict with additions, changes or deletions made by the user?